modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java

```diff
@@ -232,11 +233,18 @@ public DDMTemplate addTemplate(
 				}
 			}
 
-			if ((smallImageBytes == null) && !Validator.isUrl(smallImageURL)) {
-				smallImage = false;
+			if (smallImageBytes == null) {
+				if (smallImageURL.equals("")) {
+					throw new PortalException();
+				}
+				else if (!_isValidImageURL(smallImageURL)) {
+					throw new TemplateSmallImageInvalidURLException();
+				}
```

Comment:
I separated it because when the two fields where empty (first if statement), the message complaining about the URL doesn't made sense, so I throw a generic exception since i didn't create a specific one for this case.

Reply:
Is it possible to make the error message more generic so we can use the same exception? Thanks!